src/main/kotlin/uk/gov/justice/digital/hmpps/suicideriskformapi/service/SuicideRiskService.kt

```diff
@@ -0,0 +1,204 @@
+package uk.gov.justice.digital.hmpps.suicideriskformapi.service
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.breachnoticeapi.exception.NotFoundException
+import uk.gov.justice.digital.hmpps.suicideriskformapi.entity.AddressEntity
+import uk.gov.justice.digital.hmpps.suicideriskformapi.entity.ContactEntity
+import uk.gov.justice.digital.hmpps.suicideriskformapi.entity.SuicideRiskEntity
+import uk.gov.justice.digital.hmpps.suicideriskformapi.model.Address
+import uk.gov.justice.digital.hmpps.suicideriskformapi.model.Contact
+import uk.gov.justice.digital.hmpps.suicideriskformapi.model.CreateResponse
+import uk.gov.justice.digital.hmpps.suicideriskformapi.model.InitialiseSuicideRisk
+import uk.gov.justice.digital.hmpps.suicideriskformapi.model.SuicideRisk
+import uk.gov.justice.digital.hmpps.suicideriskformapi.repository.SuicideRiskRepository
+import java.util.*
+
+@Service
+class SuicideRiskService(
+  val suicideRiskRepository: SuicideRiskRepository,
+  @Value("\${frontend.url}") val frontendUrl: String,
+) {
+
+  @Transactional
+  fun initialiseSuicideRisk(initialiseSuicideRisk: InitialiseSuicideRisk) = suicideRiskRepository.save(
+    SuicideRiskEntity(crn = initialiseSuicideRisk.crn),
+  ).id.let {
+    CreateResponse(it, "$frontendUrl/breach-notice/$it")
+  }
+
+  fun findSuicideRiskById(id: UUID): SuicideRisk {
+    val suicideRiskEntity: SuicideRiskEntity = suicideRiskRepository.findByIdOrNull(id) ?: throw NotFoundException("SuicideRiskEntity", "id", id)
+    return suicideRiskEntity.toModel()
+  }
+
+  @Transactional
+  fun updateSuicideRisk(id: UUID, suicideRisk: SuicideRisk): SuicideRisk {
+    val suicideRiskEntity: SuicideRiskEntity = suicideRiskRepository.findByIdOrNull(id) ?: throw NotFoundException("SuicideRiskEntity", "id", id)
+    return suicideRiskRepository.save(suicideRisk.toEntity(suicideRiskEntity)).toModel()
+  }
+
+  @Transactional
+  fun deleteSuicideRisk(id: UUID): Any? {
+    val suicideRiskEntity = suicideRiskRepository.findByIdOrNull(id)
+    if (suicideRiskRepository.findByIdOrNull(id) == null) {
+      return ResponseEntity(
+        "The Suicide Risk id was not found",
+        HttpStatus.NOT_FOUND,
+      )
+    }
```

Comment:
You're calling the repository twice here - you could use `existsById()` instead
```suggestion
    if (!suicideRiskRepository.existsById(id)) throw NotFoundException("SuicideRiskEntity", "id", id)
```